src/caretogether-pwa/src/Settings/Roles/AddRole.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Button,
+  Grid,
+  TextField,
+  Checkbox,
+  FormControlLabel,
+  FormHelperText,
+  FormControl,
+} from '@mui/material';
+import { RoleDefinition } from '../../GeneratedClient';
+import { useState } from 'react';
+import { useBackdrop } from '../../Hooks/useBackdrop';
+import { api } from '../../Api/Api';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import { selectedLocationContextState } from '../../Model/Data';
+import { organizationConfigurationEdited } from '../../Model/ConfigurationModel';
+
+interface DrawerProps {
+  onClose: () => void;
+}
+
+interface AddEditCommunityDrawerProps extends DrawerProps {}
+
+export function AddRole({ onClose }: AddEditCommunityDrawerProps) {
+  const [roleName, setRoleName] = useState('');
+  const [isProtected, setIsProtected] = useState(false);
+
+  const { organizationId } = useRecoilValue(selectedLocationContextState);
+
+  const storeEdits = useSetRecoilState(organizationConfigurationEdited);
+
+  const withBackdrop = useBackdrop();
+
+  async function save() {
+    await withBackdrop(async () => {
+      const newConfig = await api.configuration.putRoleDefinition(
+        organizationId!,
+        roleName,
+        new RoleDefinition({
+          roleName,
+          isProtected,
+          permissionSets: [],
+        })
+      );
+
+      storeEdits(newConfig);
+      onClose();
+    });
+  }
+
+  return (
+    <Grid
+      container
+      spacing={2}
+      maxWidth={500}
+      component="form"
```

Comment:
Add an inline comment:
```suggestion
      component="form" // This allows easier editing & submitting via the keyboard
```